### ISSUE TITLE
Refresh CardList component

### ISSUE THREAD

**Issue by connor-baer** [sumup-oss/circuit-ui#794]
## Component to amend

👀 [View CardList in Storybook](https://circuit.sumup.com/?path=/story/components-card-cardlist--base)

### Design

The current component was created for one use case (the transaction history in SumUp's Dashboard app). Over time, it's been used in many other scenarios in inaccessible and inconsistent ways. We should review those use cases, define clear usage guidelines, and make the component adapt to its context. Some of this has already been done for [Circuit UI Mobile](https://www.figma.com/file/lgvQDus1CRKC7heaA1EzkR/Circuit-UI-Mobile?node-id=1271%3A0):

<img width="1034" alt="Screen Shot 2021-01-29 at 10 07 23" src="https://user-images.githubusercontent.com/11017722/106255059-ca241980-6219-11eb-8545-258d63611bc6.png">

### API

- The component should be renamed to ListItem since it has nothing to do with the Card component and should match the name in Circuit UI Mobile.
- The component should be migrated to TypeScript.
- The component's accessibility needs to be reviewed, particularly when navigating via keyboard or screenreader.
